bundles/org.openhab.voice.rustpotterks/src/main/java/org/openhab/voice/rustpotterks/internal/RustpotterKSService.java

```diff
@@ -196,10 +223,18 @@ private void processAudioStream(Rustpotter rustpotter, KSListener ksListener, Au
                 }
                 if (numBytesRead != remaining) {
                     remaining = remaining - numBytesRead;
+                    try {
+                        Thread.sleep(remaining / bytesPerMs);
+                    } catch (InterruptedException ignored) {
```

Comment:
Are you sure it shouldn't be ignored? E.g. when someone shuts down the runtime the thread stop running ASAP.

Reply:
Yes, I think you are right, I have added a warning to be aware when these happens and set the aborted bool to true.